helper/bufconndialer/bufconndialer.go

```diff
@@ -0,0 +1,48 @@
+// BufConnWrapper implements consul-template's TransportDialer using a
```

Comment:
Oops, no tests. I'll try to get to that later today. https://github.com/hashicorp/nomad/issues/12483